src/engine/MessageQueue.hxx

```diff
@@ -40,13 +40,19 @@ public:
   bool peek(void) noexcept;
 
   /**
-   * @details Blocks the thread until an event is received
-   * @returns an Enumerable of Events on the Queue
+   * @details Blocks the thread until an event is received.
+   * @returns an Enumerable of the Queue Events.
    * @post the Queue is emptied
    * @post Enumerable contains at least one Event
    */
   Enumerable getEnumerable(void);
 
+  /**
+   * @details Blocks the thread until an event is received or timeout occurred.
+   * @returns an Enumerable of the Queue Events if at least one event received before timeout occurred, otherwise empty enumerable of Queue Events.
```

Comment:
Can you add a `@tparam`  and `@param` explaining what Duration is?
```suggestion
   * @returns an Enumerable of the Queue Events if at least one event received before timeout occurred, otherwise empty enumerable of Queue Events.
   * @tparam Duration the type of the Duration
   * @param the duration value
```